code/modules/mining/drilling/deep_drill.dm

```diff
@@ -0,0 +1,356 @@
+	..()
+	harvest_speed = 0
+	capacity = 0
+	charge_use = 37
+	radius = RADIUS
+
+	for(var/obj/item/stock_parts/P in component_parts)
+		if(istype(P, /obj/item/stock_parts/micro_laser))
+			harvest_speed = P.rating
+		if(istype(P, /obj/item/stock_parts/matter_bin))
+			capacity = 200 * P.rating
+		if(istype(P, /obj/item/stock_parts/capacitor))
+			charge_use -= 8 * (P.rating - harvest_speed)
+			charge_use = max(charge_use, 0)
+		if(istype(P, /obj/item/stock_parts/scanning_module))
+			radius = RADIUS + P.rating
+	cell = locate(/obj/item/cell/large) in component_parts
+
+/obj/machinery/mining/deep_drill/proc/system_error(var/error)
+
+	if(error)
+		visible_message(SPAN_NOTICE("\The [src] flashes a '[error]' warning."))
+	need_player_check = TRUE
+	active = FALSE
+	if(GC)
+		GC.stop()
+		GC = null
+	update_icon()
+
+/obj/machinery/mining/deep_drill/proc/get_resource_field()
+	resource_field = list()
+	need_update_field = FALSE
+
+	var/turf/simulated/T = get_turf(src)
+	if(!istype(T))
+		return
+
+	for(var/turf/simulated/mine_trufs in range(T, radius))
+		if(mine_trufs.has_resources)
+			resource_field += mine_trufs
+
+	if(!resource_field.len)
+		system_error("resources depleted")
+
+/obj/machinery/mining/deep_drill/proc/use_cell_power()
+	if(!cell)
+		return FALSE
+	if(cell.checked_use(charge_use))
+		return TRUE
+	return FALSE
+
+/obj/machinery/mining/deep_drill/proc/check_surroundings()
+	// Check if there is no dense obstacles around the drill to avoid blocking access to it
+	for(var/turf/F in block(locate(x - 1, y - 1, z), locate(x + 1, y + 1, z)))
+		if(F != loc && F.contains_dense_objects(TRUE))
+			return TRUE
+	return FALSE
+
+/obj/machinery/mining/deep_drill/attack_generic(mob/user, var/damage)
+	user.do_attack_animation(src)
+	visible_message(SPAN_DANGER("\The [user] smashes into \the [src]!"))
+	take_damage(damage)
+	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN * 1.5)
+
+/obj/machinery/mining/deep_drill/bullet_act(var/obj/item/projectile/Proj)
+	..()
+	var/damage = Proj.get_structure_damage()
+	take_damage(damage)
+
+/obj/machinery/mining/deep_drill/proc/take_damage(value)
+	health = min(max(health - value, 0), max_health)
+	if(health == 0)
+		system_error("critical damage")
+		if(prob(10)) // Some chance that the drill completely blows up
+			var/turf/O = get_turf(src)
+			if(!O) return
+			explosion(O, -1, 1, 4, 10)
+			qdel(src)
+
+/obj/machinery/mining/deep_drill/examine(mob/user)
+	. = ..()
+	if(health <= 0)
+		to_chat(user, "\The [src] is wrecked.")
+	else if(health < max_health * 0.25)
+		to_chat(user, "<span class='danger'>\The [src] looks like it's about to break!</span>")
+	else if(health < max_health * 0.5)
+		to_chat(user, "<span class='danger'>\The [src] looks seriously damaged!</span>")
+	else if(health < max_health * 0.75)
+		to_chat(user, "\The [src] shows signs of damage!")
+
+/obj/machinery/mining/deep_drill/verb/unload()
+	set name = "Unload Drill"
+	set category = "Object"
+	set src in oview(1)
+
+	if(usr.stat)
```

Comment:
what about casting the usr as a mob , checking if its a mob , and then calling incapacitated on the mob ? its far more generalized and handles lying down , cuffs , etc , that stat does not handle.

Reply:
Will do it